### PR TITLE
Paginate flagged users and more

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Implement `UserConnection` on the graph
- Allow property `statuses` for filtering in a user query.
- Refactor `FlaggedUsers` related code
- Refactor Community styles
- Add transition effect
- Implement graphql mutation updators.
- Always refetch when switching to flagged Users.

## How to test this PR?
- Have more than 10! flagged users to see pagination ^.^
- See transition effect when removing a user
- See refetch taking place whenever switching to flagged user view.

### CODE REVIEW

File: graph/typeDefs.graphql [hunk 1]
```diff
@@ -730,8 +751,8 @@ type RootQuery {
   # role.
   me: User
 
-  # Users returned based on a query.
-  users(query: UsersQuery): [User]
+  # Users returned based on a query. Requires the `ADMIN` role.
```
Comment (wyattjoh) [inline graph/typeDefs.graphql:754]: Shouldn't need to indicate role here.

Comment (alinhle) [reply graph/typeDefs.graphql:754]: Well yeah but you need special privilege to access it, not sure how to describe that. (Just copied what we used before)

Comment (alinhle) [reply graph/typeDefs.graphql:754]: Requires the `SEARCH_OTHER_USERS` permission?

Comment (wyattjoh) [reply graph/typeDefs.graphql:754]: I would suggest we remove all the role requirements from the comments to be honest until we can get custom annotations.

Comment (alinhle) [reply graph/typeDefs.graphql:754]: Custom annotations, is that a thing?

**Review by kgardnr — APPROVED** [review]

**Pull request merged**